mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,405 @@
+            if y_probs is not None:
+                fpr, tpr, thresholds = sk_metrics.roc_curve(y, y_prob)
+                roc_curve_pandas_df = pd.DataFrame(
+                    {"fpr": fpr, "tpr": tpr, "thresholds": thresholds}
+                )
+                self._log_pandas_df_artifact(
+                    artifacts, temp_dir, roc_curve_pandas_df, run_id, "roc_curve", dataset_name, model,
+                )
+
+                roc_auc = sk_metrics.auc(fpr, tpr)
+                metrics["roc_auc"] = roc_auc
+
+                def plot_roc_curve():
+                    sk_metrics.RocCurveDisplay(
+                        fpr=fpr, tpr=tpr, roc_auc=roc_auc,
+                    ).plot()
+
+                if hasattr(sk_metrics, 'RocCurveDisplay'):
+                    self._log_image_artifact(
+                        artifacts, temp_dir, plot_roc_curve, run_id, "roc_curve", dataset_name,
+                    )
+
+                precision, recall, thresholds = sk_metrics.precision_recall_curve(y, y_prob)
+                thresholds = np.append(thresholds, [1.0], axis=0)
+                pr_curve_pandas_df = pd.DataFrame(
+                    {"precision": precision, "recall": recall, "thresholds": thresholds}
+                )
+                self._log_pandas_df_artifact(
+                    artifacts, temp_dir, pr_curve_pandas_df, run_id, "precision_recall_curve",
+                    dataset_name, model,
+                )
+
+                pr_auc = sk_metrics.auc(recall, precision)
+                metrics["precision_recall_auc"] = pr_auc
+
+                def plot_pr_curve():
+                    sk_metrics.PrecisionRecallDisplay(
+                        precision, recall,
+                    ).plot()
+
+                if hasattr(sk_metrics, 'PrecisionRecallDisplay'):
+                    self._log_image_artifact(
+                        artifacts, temp_dir, plot_pr_curve, run_id, "precision_recall_curve", dataset_name,
+                    )
+
+                self._log_image_artifact(
+                    artifacts, temp_dir,
+                    lambda: plot_lift_curve(y, y_probs),
+                    run_id, "lift_curve", dataset_name,
+                )
+
+            def plot_confusion_matrix():
+                sk_metrics.ConfusionMatrixDisplay(confusion_matrix=confusion_matrix).plot()
+
+            if hasattr(sk_metrics, 'ConfusionMatrixDisplay'):
+                self._log_image_artifact(
+                    artifacts, temp_dir, plot_confusion_matrix, run_id, "confusion_matrix", dataset_name,
+                )
+
+        self._log_metrics(run_id, metrics, dataset_name)
+
+        if evaluator_config.get('log_model_explainability', True):
+            self._log_model_explainability(
+                artifacts, temp_dir, model, X, dataset_name, feature_names, run_id, evaluator_config
+            )
+
+        return EvaluationResult(metrics, artifacts)
```

Comment:
I ran the following workload and the resulting artifacts were empty:

```
import mlflow
from mlflow.models.evaluation import EvaluationDataset

from sklearn import datasets
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split

# prepare train and test data
iris = datasets.load_iris()
X = iris.data
y = iris.target
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

with mlflow.start_run():
    rid = mlflow.active_run().info.run_id

    logistic = LogisticRegression(C=1.0)
    logistic.fit(X_train, y_train)
    mlflow.sklearn.log_model(logistic, artifact_path="mymodel")

    results = mlflow.evaluate(
        model="runs:/%s/mymodel" % rid,
        model_type="regressor",
        dataset=EvaluationDataset(X_test, y_test),
        evaluator_config={
            "log_model_explainability": False,
        },
        evaluators="default",
    )
    print(results.metrics)
    print(results.artifacts)
```

```
{'example_count': 30, 'mean_absolute_error': 0.0, 'mean_squared_error': 0.0, 'root_mean_squared_error': 0.0, 'sum_on_label': 31, 'mean_on_label': 1.0333333333333334, 'r2_score': 1.0, 'max_error': 0, 'mean_absolute_percentage_error': 0.0}
{}
```

Can we debug / fix / add a test case?

Reply:
The code model_type "regressor" is mismatch with `LogisticRegression` (it is a classifier)

In my latest code, I change `LogisticRegression` to `LinearRegression` or change model type to `classifier`, both the 2 cases latest code run correctly.

Reply:
LogisticRegression is a classifier algorithm (well, sort of... it's USED as a classifier algorithm but behaves internally like a linear solver). If the model provided does not match with the model_type that is configured for evaluation, the user should probably be alerted that the Evaluator can't, well, evaluate. 
This should be an AttributeError Exception that is thrown (or a custom Exception) since the wrong metrics will be pulled to execute and could prove very confusing for an end-user.

Reply:
>  If the model provided does not match with the model_type

How to verify this ? We can check the evaluation data labels to be integers , but not accurate.

Reply:
@WeichenXu123 @BenWilson2 I think providing a hint about the model type mismatch is helpful, but it's probably hard for us to get this right in 100% of cases. Instead of throwing an exception, can we try to detect this case and *print a warning*? A simple check would be to count the number of distinct elements in the labels; if the number of distinct elements is roughly the size of the label set, then we can infer that it's probably not a classification problem.

Reply:
the inferring logic I use:
```
def _infer_model_type_by_labels(labels):
    distinct_labels = set(labels)
    for v in distinct_labels:
        if v < 0 or not float(v).is_integer():
            return "regressor"
    if len(distinct_labels) > 1000 and len(distinct_labels) / len(labels) > 0.7:
        return "regressor"
    return "classifier"
```

Reply:
Fantastic! Great work, @WeichenXu123 !